src/analysis.zig

```diff
@@ -3660,7 +3660,25 @@ pub const DeclWithHandle = struct {
                 })) orelse return null,
                 .Single,
             ),
-            .assign_destructure => null, // TODO
+            .assign_destructure => |pay| {
+                const rhs = tree.nodes.items(.data)[pay.node].rhs;
+                const node = try analyser.resolveTypeOfNode(.{
+                    .node = tree.nodes.items(.data)[pay.node].rhs,
+                    .handle = self.handle,
+                }) orelse return null;
+                return switch (node.data) {
+                    .array => blk: {
+                        var buffer: [2]Ast.Node.Index = undefined;
+                        const array = tree.fullArrayInit(&buffer, rhs) orelse return null;
+                        break :blk try analyser.resolveTypeOfNode(.{
+                            .node = array.ast.elements[pay.index],
+                            .handle = self.handle,
+                        });
+                    },
```

Comment:
The child type of the array has already been resolved.
```suggestion
                    .array => |array_info| try array_info.elem_ty.instanceTypeVal(analyser),
```

Reply:
I should have also mentioned that this makes the `rhs` variable above unused.

Reply:
yeah just fixing that manually and testing will push in a second